tasks/handle-action-destroy.yaml

```diff
@@ -1,73 +1,36 @@
 ---
-- name: Cancel provision tower jobs
-  when: >-
-    'completeTimestamp' not in vars.anarchy_subject.status.towerJobs.provision
-  # Loop over all provision tower jobs, "*Job" keys from status
-  loop: >-
-    {{ vars.anarchy_subject.status.towerJobs.provision | default({}) | dict2items | list
-     | to_json | from_json
-     | json_query("[?ends_with(key, 'Job')]")
-    }}
-  loop_control:
-    label: "{{ _job.key }}-{{ _job.value }}"
-    loop_var: _job
-  uri:
-    url: https://{{ babylon_tower.hostname }}/api/v2/jobs/{{ _job.value }}/cancel/
-    url_username: "{{ babylon_tower.user }}"
-    url_password: "{{ babylon_tower.password }}"
-    force_basic_auth: true
-    # FIXME - Should validate certs
-    validate_certs: false
-    return_content: true
-    method: POST
-    status_code:
-    - 202
-    - 404
-    - 405
-  register: r_cancel
+- name: Cancel provision tower job
+  when:
+  - "'deployerJob' in vars.anarchy_subject.status.towerJobs.provision"
+  - "'completeTimestamp' not in vars.anarchy_subject.status.towerJobs.provision"
+  awx.awx.tower_job_cancel:
```

Comment:
ho we're back to using the awx module ?
i can't remember the reason why we didn't use it in the first place. It was somewhat limited.

Reply:
We couldn't use the awx module with Tower 3.6. It works great with 3.7+

Reply:
Also, the awx collection may not have even existed when we started out? And the core implementation of `tower_*` modules used tower cli, which the collection has fixed.

Reply:
And when we first implemented this initially, iirc, we did not have ansibleGalxy support in Anarchy.